test/views/declaration/summary/SummaryPageViewAmendSpec.scala

```diff
@@ -41,19 +44,47 @@ class SummaryPageViewAmendSpec extends SummaryPageViewSpec {
 
     behave like commonBehaviour(document)
 
-    behave like sectionsVisiblity(view)
+    behave like sectionsVisibility(view)
 
     behave like displayWarning(document)
 
     "should display correct title" in {
-      document.getElementById("title").text() mustBe messages("declaration.summary.amend-header")
+      document.getElementById("title").text mustBe messages("declaration.summary.amend-header")
     }
 
     "should display correct back link" in {
       val backButton = document.getElementById("back-link")
 
-      backButton.text() mustBe messages("site.back")
-      backButton must haveHref(controllers.routes.SubmissionsController.displayListOfSubmissions())
+      backButton.text mustBe messages("site.back")
+      backButton must haveHref(SubmissionsController.displayListOfSubmissions())
+    }
+
+    "not display a 'View Declaration Errors' button" when {
+      List(DRAFT, INITIAL, COMPLETE).foreach { status =>
+        s"the declaration is in '$status' status and" when {
+          "declaration's 'parenDeclarationId' is NOT defined" in {
+            val document = view(aDeclaration(withStatus(DRAFT)))
+            val buttons = document.getElementsByClass("govuk-button--secondary")
+            buttons.size mustBe 0
+          }
+        }
+      }
+    }
+
+    "display a 'View Declaration Errors' button" when {
+      "the declaration is in 'DRAFT' status and" when {
```

Comment:
Should you not also test for the scenario where the `parentDeclarationId` is Some but the draft dec's status is not `DRAFT` that there is no button?